usr/src/cmd/zdb/zdb.c

```diff
@@ -1111,7 +1116,7 @@ dump_history(spa_t *spa)
 	umem_free(buf, SPA_MAXBLOCKSIZE);
 
 	(void) printf("\nHistory:\n");
-	for (int i = 0; i < num; i++) {
+	for (unsigned i = 0; i < num; i++) {
```

Comment:
By default FreeBSD compiles with -Werror,-Wsign-compare which flagged this line as an error.  We've had to disable that warning for ZFS.  I think the best approach would be to slowly increase the WARNS settings for different parts of ZFS, one at a time, fixing the errors as the appear.  If OpenZFS keeps warnings disabled, then FreeBSD's compiler will occasionally complain about newly imported OpenZFS code.  But we can fix those as they pop up.

I choose zdb because FreeBSD had to disable _all_ compiler warnings to get it to build, and that masked genuine errors when I was working on other stuff in zdb, which cost me a couple of days.